File: 승준/20230629/tree_1068.py
import sys
input = sys.stdin.readline

def dfs(num, parent):
    parent[num] = -2
    for i in range(len(parent)):
        if parent[i] == num:
            dfs(i, parent)

num = int(input())
parent = list(map(int, input().split()))
delete = int(input())

dfs(delete, parent)
count = 0
for j in range(len(parent)):
    if parent[j] != -2 and j not in parent:
        count += 1
print(count) 

File: 승준/20230629/tree_1068.js
function dfs(num, parent) {
  parent[num] = -2;
  for (let i = 0; i < parent.length; i++) {
    if (parent[i] === num) {
      dfs(i, parent);
    }
  }
}

function contains(parent, num) {
  for (let n of parent) {
    if (n === num) {
      return true;
    }
  }
  return false;
}

function main() {
  const readline = require("readline");
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  rl.question("Enter the number: ", (num) => {
    const parent = new Array(num);
    rl.question("Enter the parent array: ", (parentInput) => {
      const parentTokens = parentInput.trim().split(" ");
      for (let i = 0; i < parent.length; i++) {
        parent[i] = parseInt(parentTokens[i]);
      }
      rl.question("Enter the delete node: ", (deleteInput) => {
        const deleteNode = parseInt(deleteInput);

        dfs(deleteNode, parent);
        let count = 0;
        for (let i = 0; i < parent.length; i++) {
          if (parent[i] !== -2 && !contains(parent, i)) {
            count++;
          }
        }
        console.log(count);
        rl.close();
      });
    });
  });
}

main();
